app/src/main/java/com/example/qstreak/ui/OnboardingSignupFragment.kt

```diff
@@ -29,24 +29,17 @@ class OnboardingSignupFragment : Fragment() {
             null,
             false
         )
-        binding.continueButton.setOnClickListener {
-            createUser()
+        binding.lifecycleOwner = activity
+        binding.viewModel = onboardingViewModel
+        binding.signUpCallback = {
+            // TODO data validation - zip required, age and household size optional
+            onboardingViewModel.createUser(requireContext())
         }
         observeSignup()
 
         return binding.root
     }
 
-    private fun createUser() {
-        // TODO data validation - zip required, age and household size optional
-        onboardingViewModel.createUser(
-            requireContext(),
-            binding.ageInput.editText?.text.toString().toInt(),
-            binding.householdSizeInput.editText?.text.toString().toInt(),
-            binding.zipCodeInput.editText?.text.toString()
```

Comment:
I moved this to the `ViewModel`, it can use two-way databinding to read/write the values